code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -248,78 +248,19 @@
 	gold_core_spawnable = NO_SPAWN
 	faction = list(ROLE_SYNDICATE)
 	rarechance = 10
-	/// Keeping track of the nuke disk for the functionality of storing it.
-	var/obj/item/disk/nuclear/disky
-	/// Location of the file storing disk overlays
-	var/icon/disk_overlay_file = 'icons/mob/simple/carp.dmi'
-	/// Colored disk mouth appearance for adding it as a mouth overlay
-	var/mutable_appearance/colored_disk_mouth
 
 /mob/living/simple_animal/hostile/carp/cayenne/Initialize(mapload)
 	. = ..()
 	AddElement(/datum/element/pet_bonus, "bloops happily!")
-	ADD_TRAIT(src, TRAIT_DISK_VERIFIER, INNATE_TRAIT) //carp can verify disky
-	ADD_TRAIT(src, TRAIT_CAN_STRIP, INNATE_TRAIT) //carp can take the disk off the captain
-	ADD_TRAIT(src, TRAIT_CAN_USE_NUKE, INNATE_TRAIT) //carp SMART
-
-/mob/living/simple_animal/hostile/carp/cayenne/death(gibbed)
-	if(disky)
-		disky.forceMove(drop_location())
-		disky = null
-	return ..()
-
-/mob/living/simple_animal/hostile/carp/cayenne/Destroy(force)
-	QDEL_NULL(disky)
-	return ..()
-
-/mob/living/simple_animal/hostile/carp/cayenne/examine(mob/user)
-	. = ..()
-	if(disky)
-		. += span_notice("Wait... is that [disky] in [p_their()] mouth?")
-
-/mob/living/simple_animal/hostile/carp/cayenne/AttackingTarget(atom/attacked_target)
-	if(istype(attacked_target, /obj/item/disk/nuclear))
-		var/obj/item/disk/nuclear/potential_disky = attacked_target
-		if(potential_disky.anchored)
-			return
-		potential_disky.forceMove(src)
-		disky = potential_disky
-		to_chat(src, span_nicegreen("YES!! You manage to pick up [disky]. (Click anywhere to place it back down.)"))
-		update_icon()
-		if(!disky.fake)
-			client.give_award(/datum/award/achievement/misc/cayenne_disk, src)
-		return
-	if(disky)
-		if(isopenturf(attacked_target))
-			to_chat(src, span_notice("You place [disky] on [attacked_target]"))
-			disky.forceMove(attacked_target)
-			disky = null
-			update_icon()
-		else
-			disky.melee_attack_chain(src, attacked_target)
-		return
-
-	if(istype(attacked_target, /obj/machinery/nuclearbomb))
-		var/obj/machinery/nuclearbomb/nuke = attacked_target
-		nuke.ui_interact(src)
+	var/mutable_appearance/disk = mutable_appearance('icons/mob/simple/carp.dmi', "disk_overlay")
+	var/mutable_appearance/mouth = mutable_appearance(SSgreyscale.GetColoredIconByType(/datum/greyscale_config/carp/disk_mouth, greyscale_colors), "disk_mouth")
```

Comment:
Is there any chance that we can defer this until it's necessary without making the code hard to read? GAGS has a non-negligible impact on init times.

Reply:
The only way I could think to do this is to delegate the overlays back to the mob again, as I can't think of a good way to pass GAGS info through to the component without just assembling it
Not sure if this is actually a good idea, makes it hard for someone to set up via VV too but I'm not honestly sure people doing that were going to bother giving it graphics anwyay

Reply:
Yeah I can't think of a great way myself, I'll just let this go then